src/Data/Connection/AbstractConnectionResolver.php

```diff
@@ -257,22 +261,29 @@ abstract public function get_query_args();
 	abstract public function get_query();
 
 	/**
-	 * Should_execute
-	 *
-	 * Determine whether or not the query should execute.
-	 *
-	 * Return true to execute, return false to prevent execution.
-	 *
-	 * Various criteria can be used to determine whether a Connection Query should
-	 * be executed.
+	 * Used to determine whether the connection query should be executed. This is useful for short-circuiting the connection resolver before executing the query.
 	 *
-	 * For example, if a user is requesting revisions of a Post, and the user doesn't have
-	 * permission to edit the post, they don't have permission to view the revisions, and therefore
-	 * we can prevent the query to fetch revisions from executing in the first place.
+	 * When `pre_should_excecute()` returns false, that's a sign the Resolver shouldn't execute the query. Otherwise, the more expensive logic logic in `should_execute()` will run later in the lifecycle.
 	 *
-	 * @return bool
+	 * @param mixed                                $source  Source passed down from the resolve tree
+	 * @param array<string,mixed>                  $args    Array of arguments input in the field as part of the GraphQL query.
+	 * @param \WPGraphQL\AppContext                $context The app context that gets passed down the resolve tree.
+	 * @param \GraphQL\Type\Definition\ResolveInfo $info    Info about fields passed down the resolve tree.
 	 */
-	abstract public function should_execute();
+	protected function pre_should_execute( $source, array $args, AppContext $context, ResolveInfo $info ): bool {
+		$should_execute = true;
+
+		/**
+		 * If the source is a Post and the ID is empty, we should not execute the query.
+		 *
+		 * @todo this can probably be moved to the PostObjectConnectionResolver when we don't care about b/c.
```

Comment:
I don't think this can move to the PostObjectConnectionResolver because the `$source` is the node the connection resolver is coming _from_ not going to. 

So a `Post->User` connection which would use the `UserConnectionResolver` needs this, etc. 

We could probably generalize it better to validate that the `$source` is valid in some way to include this specific scenario and others. I can't recall the intent of this and what edge case we faced where we were passing `Post` without an `ID` 🤔. . .but I assume we could generalize this to better validate any source 🤔

Reply:
Updated in https://github.com/wp-graphql/wp-graphql/pull/3104/commits/8014e2c93d9a612a79998fd09b0a52786f57e5d4